blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala

```diff
@@ -91,6 +104,7 @@ class BlazeBuilder(
         }
 
       case None =>
+        if (http2Support) logger.warn("Http2 support requires TLS.")
```

Comment:
@rossabaker, here I fall back to http/1.1 if TLS is not available. This sort of conflicts with the ```isHttp2``` naming discussed above. I'm not certain we should fall back but instead error. If we do fall back, maybe ```isHttp2``` isn't appropriate and something more like ```enableHttp2``` is better.